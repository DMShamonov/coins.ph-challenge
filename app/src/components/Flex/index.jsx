// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PropsType } from './types.js.flow';

export default function Flex({
  className,
  children,
}: PropsType): React.Node {
  return (
    <div className={classnames('flex', className)}>
      {children}
    </div>
  );
}
