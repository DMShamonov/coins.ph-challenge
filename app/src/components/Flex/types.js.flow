// @flow

import type { Node } from 'react';

export type PropsType = {|
  className?: string,
  children: Node,
|};
