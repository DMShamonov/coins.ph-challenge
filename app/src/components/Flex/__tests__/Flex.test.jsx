import React from 'react';
import { shallow } from 'enzyme';

import Flex from '../index';

describe('`Flex` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Flex>__CONTENT__</Flex>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<Flex className="__CLASSNAME__">__CONTENT__</Flex>)).toMatchSnapshot();
  });
});
