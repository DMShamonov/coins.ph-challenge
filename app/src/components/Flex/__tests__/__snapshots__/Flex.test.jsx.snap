// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Flex\` component should render correctly with \`className\` prop 1`] = `
<div
  className="flex __CLASSNAME__"
>
  __CONTENT__
</div>
`;

exports[`\`Flex\` component should render correctly with default props 1`] = `
<div
  className="flex"
>
  __CONTENT__
</div>
`;
